message.go

```diff
@@ -175,3 +175,14 @@ func (messages *messages) copyInternal() []Message {
 	}
 	return copiedMessages
 }
+
+// Returns all messages and removes them at the same time
+func (messages *messages) Purge() []Message {
+	messages.Lock()
+	defer messages.Unlock()
+
+	copiedMessages := messages.copyInternal()
+	messages.items = nil
```

Comment:
What will happen in case when we have active SMTP session and in this time we run `server.MessagesAndPurge()`?

Reply:
Nothing special. We now have locks around modifying messages, and active SMTP session message is currently appended to messages only after the session (or more precisely, the message) is finished processing. So you will just get empty messages back. Once session finishes, the message gets appended. And next call returns the messages and clears the slice.

Reply:
Should we cover this behavior by tests (to insure explicitly that `MessagesAndPurge` does not break anything)? What are you thinking? 👨‍💻

Reply:
> the test would only have to be that message is appended at the end of SMTP session handling and not before

Yeah, it can valid test for `server.handleSession()`. Let it leave as it is. I'll add some testcases before release. Anyway thanks, great job!